packages/mermaid/src/diagrams/pie/pieRenderer.ts

```diff
@@ -125,12 +127,81 @@ export const draw: DrawDefinition = (text, id, _version, diagObj) => {
     .style('text-anchor', 'middle')
     .attr('class', 'slice');
 
-  group
+  const titleText = db.getDiagramTitle();
+  const maxAvailableWidth = pieWidth - MARGIN;
+  let fontSize = 25;
+  const minFontSize = 8;
+
+  // Start wrapping immediately
+  let wrappedTitle = wrapLabel(titleText, maxAvailableWidth, {
+    fontSize,
+    fontFamily: 'Arial',
+    fontWeight: 400,
+    joinWith: '<br/>',
+  });
+
+  let lines = wrappedTitle.split('<br/>');
+
+  // Create temporary text to measure
+  let tempTitle = svg
     .append('text')
-    .text(db.getDiagramTitle())
-    .attr('x', 0)
-    .attr('y', -(height - 50) / 2)
-    .attr('class', 'pieTitleText');
+    .attr('x', pieWidth / 2)
+    .attr('y', 30)
+    .attr('class', 'pieTitleText')
+    .style('text-anchor', 'middle')
+    .style('white-space', 'pre-line')
+    .style('font-size', fontSize + 'px');
+
+  lines.forEach((line, idx) => {
+    tempTitle
+      .append('tspan')
+      .attr('x', pieWidth / 2)
+      .attr('dy', idx === 0 ? 0 : '1.2em')
+      .text(line);
+  });
+
+  let bbox = (tempTitle.node() as SVGGraphicsElement)?.getBBox();
+
+  // Shrink if even the wrapped version is too wide
+  while (bbox && bbox.width > maxAvailableWidth && fontSize > minFontSize) {
+    fontSize -= 1;
+    tempTitle.remove();
+
+    // Re-wrap at smaller font size
+    wrappedTitle = wrapLabel(titleText, maxAvailableWidth, {
+      fontSize,
+      fontFamily: 'Arial',
+      fontWeight: 400,
+      joinWith: '<br/>',
```

Comment:
This is repeated from above, please move them into a different variable, so we only need to adjust in one place. 

<img width="526" alt="image" src="https://github.com/user-attachments/assets/c6812206-a3f7-4c04-859d-c83bcf5baf72" />